Reshape introduction and welcome episodes (#26)
esciencecenter-digital-skills/Natural-language-processing: episodes/00-welcome.md

:::::: questions
- Who is this lesson for?
- What will be covered in this lesson?
- Which datasets will be used?
::::::

:::::: objectives
- Identify the target audience
- Identify the learning goals of the lesson
- Get data that will be used in the course
::::::

# Welcome

> This course covers core concepts of Natural Language Processing (or NLP) and it is designed to equip researchers in the humanities and social sciences with the foundational skills and knowledge needed to carry over text-based research projects. 

Reviewer: ```suggestion
This lesson covers core concepts of Natural Language Processing (or NLP) . It will equip you with the foundational skills and knowledge needed to carry over text-based research projects. The lesson is designed specifically with researchers in the humanities and social sciences in mind, but is also applicable to other fields of research.
```

I think even though you focus on SSH, the lesson will still be useful for others.

Reviewer: Also, in carpentries lesson material it is common to refer to `lesson` instead of `course`

Author: Great, this answers a lot of questions and discussions Laura and I had re- lessons vs courses. Also I like that we can draw parallels across domains, beyond SSH

### What is NLP?

> Natural language processing (NLP) is an area of research and application that focuses on making natural (i.e., human) language accessible to computers so that they can be used to perform useful tasks. Research in NLP is highly interdisciplinary, drawing on concepts from computer science, linguistics, logic, mathematics, psychology, etc. In the past decade, NLP has evolved significantly with advances in technology to the point that it has become embedded in our daily lives: automatic language translation or chatGPT are only some examples. 

Reviewer: ```suggestion
Natural language processing (NLP) is an area of research and application that focuses on making natural (i.e., human) language accessible to computers so that they can be used to perform useful tasks. Research in NLP is highly interdisciplinary, drawing on concepts from computer science, linguistics, logic, mathematics, psychology, etc. In the past decade, NLP has evolved significantly with advances in technology, especially in the field of deep learning, to the point that it has become embedded in our daily lives: automatic language translation or chatGPT are only some examples. 
```

Reviewer: I feel like you should mention deep learning here since so much of NLP research and applications are actually deep learning applied to NLP.

Author: lovely addition, thank you!

:::::::::::: challenge 
## NLP in the real world
Many of the concepts and problems we will tackle in this NLP course have a tangible impact in your daily life. Let's see how.

Your task is to name three to five tools/products that you use on a daily basis and that you think leverage NLP techniques. To solve this exercise you can get some help from the web.

:::::: solution
These are some of the most popular NLP-based products that we use on a daily basis:

- Voice-based assistants (e.g., Alexa, Siri, Cortana)
- Machine translation (e.g., Google translate, Amazon translate)
- Search engines (e.g., Google, Bing, DuckDuckGo)
- Keyboard autocompletion on smartphones
- Spam filtering
- Spell and grammar checking apps
::::::
::::::::::::

The exercise above tells us that a great deal of NLP techniques is embedded in our daily life. What is that is working under the hood of these technologies?

## What will we be covering in this lesson?

> This lesson provides a high-level introduction to NLP with particular emphasis on applications in the humanities and the social sciences. We will focus on solving particular problems over the lesson. Some problems deal with capturing the meaning of a word (e.g., `head`) and how this changes over time and context (e.g., `top part of the body` vs `leaders of others`), others with identifying key entities in text (such as people, places, companies, dates and more) in literary texts and labeling each one of them with the right category name. These problems are examples of useful applications in your own research, however, they also offer a window in the latest NLP advancements that are now embedded in our daily life.

Reviewer: I think the way this part is now, it is a bit like: this gives you an idea of the kind of problems we will tackle, but you can forget everything you just read because we will get back to it anyway. Is that what you intended?

If not you could consider making short subsections and actually naming the problem (word2vec, named entity recognition). And mention the sections in which you will actually cover this.

Reviewer: But in a way, the `syllabus` already introduces what is discussed in the different sections. So maybe this information is redundant?

After following this lesson, learners will be able to:

- Explain and differentiate what are the core topics in NLP
- Identify what kinds of tasks NLP techniques excel at, and what are their limitations
- Structure a typical NLP pipeline

> - Extract vector representations of individual words, visualise it and manipulate it

Reviewer: ```suggestion
- Extract vector representations of individual words, visualise and manipulate them
```

- Applying a machine learning algorithm to textual data to extract and categorise names of entities (e.gs., places, people)
- Using natural language to produce a desired response from a large language model (LLM), i.e. prompt engineering

> - Other?

Reviewer: ```suggestion

```

> ## Datasets

Reviewer: I think this should be part of the setup instructions. Or do you have a good reason for not putting it in there?

Author: I think my idea was (similarly to [the intro to deep learning](https://carpentries-incubator.github.io/deep-learning-intro/) to provide people with some level of context re- datasets and links to download it so I think I'll make this explicit

We will be using the following material.

For the episode 01: preprocessing and word embeddings (Word2Vec):

- An excerpt from a Dutch journal of your choice dated from the 1950 up until 1989. This can be freely downloaded as a `txt` from [Delpher](https://www.delpher.nl/nl/kranten)
- Word2Vec models trained on 6 national Dutch newspaper data spanning a time period from 1950 to 1989 (Wevers, M., 2019). These models are available on [Zenodo](https://zenodo.org/records/3237380).

> For the episode 02: Transformers (BERT)

Reviewer: ```suggestion

```
Maybe a minor technical principality: I would suggest to track 'what still need to be added' in issues, so that the lesson is you always a viable product.
